### PR TITLE
Enable Softlinking/LazyLoading for NVRTC/RAND 

### PR DESCRIPTION
**Pull request by AdrianAbeyta**
Re-Opening PR:

This PR introduces changes that enable softlinking/lazy loading with CuPy v13 for curand and nvrtc

cc: @pnunna93

### CODE REVIEW

File: cupy_backends/cuda/libs/nvrtc.pyx [hunk 1]
```diff
@@ -78,176 +83,226 @@ cpdef tuple getSupportedArchs():
     check_status(status)
     return tuple(archs)
 
+>>>>>>> rocm6.1_internal_testing
```
Comment (pnunna93) [inline cupy_backends/cuda/libs/nvrtc.pyx]: seems like there is a left over from merge conflict, please delete this.

Comment (AdrianAbeyta) [reply cupy_backends/cuda/libs/nvrtc.pyx]: Yes, working on resolving some compiler issues.

**Comment by AdrianAbeyta** [comment]
Passing CI: https://github.com/ROCmSoftwarePlatform/cupy/actions/runs/6884950139

**Comment by AdrianAbeyta** [comment]
Passing UT: 
[nvrtc_softlink_test.log](https://github.com/ROCmSoftwarePlatform/cupy/files/13380582/nvrtc_softlink_test.log)

**Review by pnunna93 — APPROVED** [review]
LGTM

**Pull request merged**